packages/experiments/src/components/MicroFeedback/MicroFeedback.view.tsx

```diff
@@ -0,0 +1,128 @@
+    followUpOption: Button,
+    followUpOptionText: Text,
+    thanksContainer: Callout
+  });
+
+  const followUpOptionTokens: IButtonTokens = { contentPadding: '6px 0px' };
+
+  const onRenderFollowup = (followUp: IMicroFeedbackQuestion, targetRef: HTMLDivElement | null): JSX.Element => {
+    const onRenderCalloutItem = (item: string, index: number | undefined): JSX.Element => {
+      const listOption = (): void => {
+        onCalloutDismiss();
+        if (sendFollowUpIndex && index !== undefined && followUp) {
+          sendFollowUpIndex(followUp!.id, index);
+          onThanksShow();
+        }
+      };
+
+      return (
+        <Slots.followUpOption data-is-focusable={true} onClick={listOption} tokens={followUpOptionTokens}>
+          <Slots.followUpOptionText>{`${item}`}</Slots.followUpOptionText>
+        </Slots.followUpOption>
+      );
+    };
+
+    return (
+      <Slots.followUpContainer gapSpace={0} onDismiss={onCalloutDismiss} role="alertdialog" setInitialFocus={true} target={targetRef}>
+        <FocusZone direction={FocusZoneDirection.vertical}>
+          <Slots.followUpQuestion block variant="small">
+            {followUp.question}
+          </Slots.followUpQuestion>
+          <Slots.followUpOptionList items={followUp.options} onRenderCell={onRenderCalloutItem} />
+        </FocusZone>
+      </Slots.followUpContainer>
+    );
+  };
+
+  const showThanks = (followUp: IMicroFeedbackQuestion | undefined, voteType: string): void => {
+    // If vote while thanks is showing, dismiss
+    if (isThanksVisible) {
+      onThanksDismiss();
+    }
+
+    // Show thanks if there is no follow up question and not  unselecting a vote
+    if (!followUp && vote !== voteType) {
+      onThanksShow();
+    }
+  };
+
+  const likeVoteClick = (): void => {
+    showThanks(likeQuestion, 'like');
+    onLikeVote();
+  };
+
+  const dislikeVoteClick = (): void => {
+    showThanks(dislikeQuestion, 'dislike');
+    onDislikeVote();
+  };
+
+  return (
+    <Slots.root>
+      <Slots.iconContainer horizontal>
+        {children}
+        <div ref={likeRef}>
+          <IconButton menuIconProps={{ iconName: likeIcon }} title={likeIconTitle} onClick={likeVoteClick} />
+        </div>
+        <div ref={dislikeRef}>
+          <IconButton menuIconProps={{ iconName: dislikeIcon }} title={dislikeIconTitle} onClick={dislikeVoteClick} />
+        </div>
+      </Slots.iconContainer>
+      {likeQuestion && !hideLikeCallout && onRenderFollowup(likeQuestion, likeRef.current)}
+      {dislikeQuestion && !hideDislikeCallout && onRenderFollowup(dislikeQuestion, dislikeRef.current)}
+      {thanksText && isThanksVisible && (
+        <Slots.thanksContainer
+          setInitialFocus={false}
+          target={vote === 'like' ? likeRef.current : dislikeRef.current}
+          gapSpace={0}
+          isBeakVisible={false}
+          onDismiss={onThanksDismiss}
+        >
+          <Slots.followUpOptionText block variant="small">
```

Comment:
I wonder, do we really need a slot for both the text and the container? If so, I also wonder if this should be named `thanksText`, but it would feel like we have too many slots.

Reply:
Let me see if I can use the followup container for both.

Reply:
Didn't work well. So I let it be for now.